runtime/security.go

```diff
@@ -23,7 +23,7 @@ var ErrForbidden = errors.New("action not allowed")
 // SecurityClaims represents contextual information for the enforcement of security rules.
 type SecurityClaims struct {
 	// UserAttributes about the current user (or service account). Usually exposed through templating as {{ .user }}.
-	UserAttributes map[string]any
+	UserAttributes map[string]any `yaml:"user_attributes"`
 	// AdditionalRules are optional security rules to apply *in addition* to the built-in rules and the rules defined on the requested resource.
 	// These are currently leveraged by the admin service to enforce restrictions for magic auth tokens.
 	AdditionalRules []*runtimev1.SecurityRule
```

Comment:
Can we remove the `yaml` parsing from this type (can be handled with a separate type in the tests)? It makes it look like this type can be serialized/deserialized to YAML, which it can't (because the proto type `runtimev1.SecurityRule` is not YAML compatible)

Reply:
I wonder why we cannot do it YAML compatible, more copies of structs - more structs to maintain - harder to maintain when 2 structs are almost equal (finding the distinction is harder) - more error-prone.

Reply:
As written above, the `*runtimev1.SecurityRule` proto-generated type is not safe for use with YAML, so we don't want to give the impression that the `SecurityClaims` type is YAML-serializable. Additionally, it's just a bad practice to hack an internal type that used in many places for a single specific use case (in this case, test fixtures) – it creates unnecessary coupling.

In this case, `UserAttributes` is the only field we need to be able to populate from the tests, so it is simple to parse it from the test fixture and create a `SecurityClaims` value for it.